### PR TITLE
feat: add missing props from react-native-screens to native-stack

### PR DESCRIPTION
**Pull request by hirbod**
**Motivation**
This PR adds some missing props from `react-native-screens` to `react-navigation/native-stack`.

Especially the relevant props for the StatusBar are important if you want to use the StatusBar and NavigationBar management of RNS.

I thought it would be easiest if I provided all the compatible props in a PR, which is why I included transitionDuration. This does supersede #10454

New props:
- homeIndicatorHidden (ios)
- navigationBarColor (android)
- navigationBarHidden (android)
- statusBarColor (android)
- statusBarTranslucent (android)
- gestureDirection (ios)
- animationDuration (ios)

I made sure that I only included props which work with Expo (RNS 3.11.1, Expo SDK 45), thats why I did not include 
- hideKeyboardOnSwipe
- gestureResponseDistance

**Test plan**

All tests passed.
- yarn typescript
- yarn lint
- yarn test
- `yarn example start` runs

### CODE REVIEW

File: packages/native-stack/src/views/NativeStackView.native.tsx
Comment (WoLewicki) [inline packages/native-stack/src/views/NativeStackView.native.tsx:212]: What about the logic of `swipeDirection` that sets other props correctly if not applied? (see https://github.com/software-mansion/react-native-screens/pull/1260/files#diff-d97b3df9fc6a0f0846507a3c189b483ac83a59c21db8788fc8d08449fbeb7343R176) I think it should be added here too.

File: packages/native-stack/src/types.tsx
Comment (WoLewicki) [inline packages/native-stack/src/types.tsx:392]: Could we rename it here to `gestureDirection` to align with [JS stack](https://reactnavigation.org/docs/stack-navigator/#gesturedirection)?

File: packages/native-stack/package.json
Comment (WoLewicki) [inline packages/native-stack/package.json:55]: Could you bump the `react-native-screens` in all packages, not only here?

**Review by WoLewicki — COMMENTED** [review]
I think it is OK to add all the missing props and it will just have no effect if someone uses Expo SDK.

File: packages/native-stack/src/types.tsx
Comment (hirbod) [reply packages/native-stack/src/types.tsx:392]: But this will need changes upstream as well, no? Or you want to alias this?

File: packages/native-stack/src/views/NativeStackView.native.tsx
Comment (WoLewicki) [inline packages/native-stack/src/views/NativeStackView.native.tsx:213]: I think it would be better to call it `animationDuration` here since the main prop is called `animation` so we have all props regarding animating between screens close.

**Comment by hirbod** [comment]
> I think it is OK to add all the missing props and it will just have no effect if someone uses Expo SDK.

This will require that I install RNS 3.13.1, because some types are missing in 3.11.1, which might have negative side effects in Expo Go, or am I wrong?

File: packages/native-stack/src/types.tsx
Comment (WoLewicki) [reply packages/native-stack/src/types.tsx:392]: Nope, you can do it the same as with e.g. [`orientation`](https://github.com/react-navigation/react-navigation/blob/0b01cb0412d8c2cbc6538b0451dd1856474b7276/packages/native-stack/src/views/NativeStackView.native.tsx#L206)

File: packages/native-stack/src/views/NativeStackView.native.tsx
Comment (hirbod) [reply packages/native-stack/src/views/NativeStackView.native.tsx:213]: So the change would be

`transitionDuration={animationDuration)`, right?

File: packages/native-stack/src/types.tsx
Comment (hirbod) [reply packages/native-stack/src/types.tsx:392]: Ok, thats what I thought.

File: packages/native-stack/src/views/NativeStackView.native.tsx
Comment (WoLewicki) [reply packages/native-stack/src/views/NativeStackView.native.tsx:213]: Yes

**Comment by hirbod** [comment]
Looks like `hideKeyboardOnGesture` is not released yet upstream. Same for `gestureResponseDistance`
@WoLewicki 

I bumped to 3.13.1

**Comment by WoLewicki** [comment]
Hmm if those props are not present there, I think it is ok to stay on `3.11.0` since there is no difference in props between that version and `3.13.1` then, yeah?

**Comment by hirbod** [comment]
Correct. hideKeyboardOnSwipe and gestureResponseDistance are only available in main branch right now, therefore I can't import `ScreenProps['gestureResponseDistance']`

**Comment by WoLewicki** [comment]
We are planning on releasing new version soon, but we need to align some Fabric things first so I don't want to provide the exact date now. I think we can just submit another PR when it is done in order not to block this one.

**Comment by hirbod** [comment]
Ok, gonna finish your reviews now, will be done in 3 minutes

File: packages/native-stack/src/views/NativeStackView.native.tsx
Comment (hirbod) [reply packages/native-stack/src/views/NativeStackView.native.tsx]: Done

File: packages/native-stack/package.json
Comment (hirbod) [reply packages/native-stack/package.json:55]: Done

File: packages/native-stack/src/types.tsx
Comment (hirbod) [reply packages/native-stack/src/types.tsx]: Done

**Comment by hirbod** [comment]
Alright, all requests done @WoLewicki

**Review by WoLewicki — APPROVED** [review]
Looks good to me 🚀  Thanks for the contribution! Let's now wait for @satya164 review and I think it is ready to be merged then :tada:

**Comment by hirbod** [comment]
You're welcome, thanks for reviewing!

File: yarn.lock
Comment (kirillzyusko) [inline yarn.lock:17621]: Can't 3.4.0 and 3.11.1 be casted to a single 3.13.1?

Just curious why before we had three different libraries pointing to single library version, and now it seems we have 3 different versions of react-native-screens 🤔

Comment (hirbod) [reply yarn.lock:17621]: These are some typical monorepo issues. But we should not 3.13.1, since 3.11.1 is the version shipped with the latest Expo SDK 45. I am not sure where RNS 3.4.0 is coming from, usually this issues disappear when you delete the yarn.lock and reinstall it

Comment (hirbod) [reply yarn.lock:17621]: @WoLewicki 

```
➜  react-navigation git:(main) ✗ yarn why react-native-screens
yarn why v1.22.10
[1/4] 🤔  Why do we have the module "react-native-screens"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "react-native-screens@3.11.1"
info Has been hoisted to "react-native-screens"
info Reasons this module exists
   - "workspace-aggregator-37e960f9-a2b1-4db0-b892-7f2b0e76bc0e" depends on it
   - Hoisted from "_project_#@react-navigation#bottom-tabs#react-native-screens"
   - Hoisted from "_project_#@react-navigation#drawer#react-native-screens"
   - Hoisted from "_project_#@react-navigation#stack#react-native-screens"
   - Hoisted from "_project_#@react-navigation#example#react-native-screens"
info Disk size without dependencies: "1.84MB"
info Disk size with unique dependencies: "1.92MB"
info Disk size with transitive dependencies: "1.92MB"
info Number of shared dependencies: 2
=> Found "@react-navigation/native-stack#react-native-screens@3.13.1"
info This module exists because "_project_#@react-navigation#native-stack" depends on it.
info Disk size without dependencies: "2.25MB"
info Disk size with unique dependencies: "2.33MB"
info Disk size with transitive dependencies: "2.33MB"
info Number of shared dependencies: 2
=> Found "react-native-reanimated#react-native-screens@3.10.2"
info This module exists because "_project_#@react-navigation#drawer#react-native-reanimated" depends on it.
info Disk size without dependencies: "1.78MB"
info Disk size with unique dependencies: "1.86MB"
info Disk size with transitive dependencies: "1.86MB"
info Number of shared dependencies: 2
✨  Done in 1.19s.
```

I'll try to regenerate the lock file

Comment (hirbod) [reply yarn.lock:17621]: But another issue is that react-native-reanimated adds a peer to 3.10.2. 
Should we add a yarn resolution to package.json ?

Comment (WoLewicki) [reply yarn.lock:17621]: @tomekzaw @piaskowyk is there a reason for this peer dep in `reanimated`?

Comment (hirbod) [reply yarn.lock:17613]: I fixed all the package.json @WoLewicki 
The last piece is reanimated.

```
➜  react-navigation git:(main) yarn why react-native-screens
yarn why v1.22.10
[1/4] 🤔  Why do we have the module "react-native-screens"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "react-native-screens@3.11.1"
info Has been hoisted to "react-native-screens"
info Reasons this module exists
   - "workspace-aggregator-7ec20abb-c574-406e-97c1-14d823890d28" depends on it
   - Hoisted from "_project_#@react-navigation#bottom-tabs#react-native-screens"
   - Hoisted from "_project_#@react-navigation#drawer#react-native-screens"
   - Hoisted from "_project_#@react-navigation#native-stack#react-native-screens"
   - Hoisted from "_project_#@react-navigation#stack#react-native-screens"
   - Hoisted from "_project_#@react-navigation#example#react-native-screens"
info Disk size without dependencies: "1.84MB"
info Disk size with unique dependencies: "1.92MB"
info Disk size with transitive dependencies: "1.92MB"
info Number of shared dependencies: 2
=> Found "react-native-reanimated#react-native-screens@3.10.2"
info This module exists because "_project_#@react-navigation#drawer#react-native-reanimated" depends on it.
info Disk size without dependencies: "1.78MB"
info Disk size with unique dependencies: "1.86MB"
info Disk size with transitive dependencies: "1.86MB"
info Number of shared dependencies: 2
✨  Done in 1.24s.
```

Comment (kirillzyusko) [reply yarn.lock:17613]: @hirbod looks better 👍 But why ^3.4.0 is pointed to 3.10.2 now, whereas we have 3.11.1 in the project? I thought ^3.4.0 means 3.x library (according to npm sematic versioning), so it should re-use 3.11.1 version? Or I'm missing something?

Comment (hirbod) [reply yarn.lock:17613]: Not sure. 3.4.0 is not being hoisted regarding to yarn why. It's 3.11.1 and 3.10.2 because of rea

**Review by kirillzyusko — APPROVED** [review]

File: yarn.lock
Comment (hirbod) [reply yarn.lock:17613]: @WoLewicki looks like the peer has been removed in 2.8.0
https://github.com/software-mansion/react-native-reanimated/commit/5601bc62cd5e1e2b4a1c41ce3e9d7059207a37d9

Shall I bump REA to 2.8.0? It is also the shipped version with SDK 45

Comment (WoLewicki) [reply yarn.lock:17613]: Yeah, I think it is a good idea to bump it too if nothing breaks then

Comment (hirbod) [reply yarn.lock:17613]: @WoLewicki upgrading REA to 2.8.0 fixes the issues but the example app will stuck on SplashScreen. Not sure if thats really the scope of this PR for me to upgrade to Expo SDK 45 now? I could do this in another PR

Comment (WoLewicki) [reply yarn.lock:17613]: That's weird. Does the app crash then?

Comment (hirbod) [reply yarn.lock:17613]: It does look like that. It just stuck on the SplashScreen as soon as I bump REA to 2.8.0. Guess some work has to be done (upgrading to Expo 44). I think there had been changes with auto installing which don't work with SDK 44

**Comment by brentvatne** [comment]
can you open a pr to the website repo (https://github.com/react-navigation/react-navigation.github.io) to update the docs for this as well?

**Comment by hirbod** [comment]
Sure, but might need a couple of days since I am caught up with work currently

**Comment by hirbod** [comment]
@brentvatne done.
https://github.com/react-navigation/react-navigation.github.io/pull/1151

Please feel to correct the text directly if you're not happy with wording.

**Comment by Chasty** [comment]
are you going to merge this PR? @hirbod maybe we need to update react native screens to the latest version.

`"react-native-screens": "^3.13.1"`

**Comment by hirbod** [comment]
@brentvatne any chance to get some traction here? :)

**Comment by kacperkapusciak** [comment]
Hi @hirbod 👋 
Would you mind resolving conflicts on this PR as I don't have permissions to push to forks?

Thanks

**Comment by hirbod** [comment]
Hi @kacperkapusciak,

I resolved the conflicts. Once the checks are done we should be good to go :)

File: packages/native-stack/src/views/NativeStackView.native.tsx
Comment (kacperkapusciak) [inline packages/native-stack/src/views/NativeStackView.native.tsx:219]:
```suggestion
      homeIndicatorHidden={autoHideHomeIndicator}
```
After discussing with @satya164 we've decided to name this `autoHideHomeIndicator` as it's impossible to hide it permanently. It just prefers to stay hidden.

File: packages/native-stack/src/types.tsx
Comment (kacperkapusciak) [inline packages/native-stack/src/types.tsx:383]: This prop doesn't seem to work at all in my emulator neither in expo managed nor bare workflow. Maybe something with my device but could you double-check this? 🙏 

I'll test try to test this more

**Review by kacperkapusciak — CHANGES_REQUESTED** [review]
Is it possible for you to allow maintainers to make changes to this branch? That would help me greatly 🙏 

Thanks!

**Comment by hirbod** [comment]
@kacperkapusciak can you tell me where I can activate this?

File: packages/native-stack/src/views/NativeStackView.native.tsx
Comment (hirbod) [reply packages/native-stack/src/views/NativeStackView.native.tsx:219]: Ok, keep in mind that we need to update this in my documentation PR as well.

File: packages/native-stack/src/types.tsx
Comment (hirbod) [reply packages/native-stack/src/types.tsx:383]: It was working for when I tested it, but only when you're not using StatusBar from react-native or expo IIRC

File: yarn.lock
Comment (hirbod) [reply yarn.lock:18212]: Not 2, even 3 :D

Comment (hirbod) [reply yarn.lock]: Should be resolved now

**Comment by cenzovit** [comment]
Any update on getting the remaining missing props into native-stack? Looks like react-native-screens has since released versions with the final missing props added.

**Comment by hirbod** [comment]
Yeah, we could make a new PR with

```
hideKeyboardOnSwipe
gestureResponseDistance
```

**Comment by BlueBazze** [comment]
Might not be an issue. And not sure if i have messed anything up.
Using the option `navigationBarColor`.
If its set to an RGB value, the app crashes with the message `java.lang.RuntimeException: java.lang.reflect.InvocationTargetException`
If its set to an RGBA value, its works as 

Navigating to a screen with that option, it is as if the navigationBar is being set to position: absolute
Meaning that any content that is positioned at the bottom of your screen is now located behind the navigationBar.

Also using the option `navigationBarHidden`
Navigating to a screen with this option, it looks fine at first. But after the first tap anywhere on the screen causes the navigationBar to appear again.

**Comment by hirbod** [comment]
@BlueBazze you should file this as issue in the react-native-screens repo with a reproduction.

**Pull request merged**
